Messenger/AppDelegate.mm

```diff
@@ -230,4 +230,22 @@ - (void)webView:(WebView *)sender decidePolicyForNewWindowAction:(NSDictionary *
   [listener ignore];
 }
 
+- (void)webView:(WebView *)sender runOpenPanelForFileButtonWithResultListener:(id < WebOpenPanelResultListener >) resultListener
+{
+    // Create the File Open Dialog class.
+    NSOpenPanel* openDlg = [NSOpenPanel openPanel];
+    
+    // Enable the selection of files in the dialog.
+    [openDlg setCanChooseFiles:YES];
+    
+    // Enable the selection of directories in the dialog.
+    [openDlg setCanChooseDirectories:YES];
```

Comment:
Would this make it so you can upload directories?  The web version doesn't let you do this.